src/Select.tsx

```diff
@@ -81,6 +84,7 @@ const Select = React.forwardRef<HTMLSelectElement, SelectProps>(
             fill: disabled ? 'GrayText' : 'FieldText',
           },
         },
+        ...sx,
```

Comment:
Hi! This is 99% there.

`...sx` does a shallow merge with the existing styles so it might accidentally override nested style objects. We have a `merge` utility for sx that deep merges. [Here is a another component that uses merge for reference](https://github.com/primer/react/blob/main/src/ActionList/Divider.tsx#L17-L25).

---

For example, with shallow merge:

```tsx
<Select sx={{
 '@media screen and (forced-colors: active)': {
    color: 'fg.muted'
 }
}}>
```

`...props.sx` will completely overwrite the '@media screen and (forced-colors: active)' key and remove svg styles

Reply:
Thanks! Makes sense!